.idea/Working version/script.js

```diff
@@ -335,30 +336,50 @@ function zoomed() {
 //this function is called whenever the data are changed.
 
 function updateComplex(newValue) {
-    //update coverage radius and recompute complexes
 
+    complexSVG.attr('cursor', 'wait')
 
-    //update slider value and/or tex value
-    complexRadius=+newValue;
+    //update coverage radius and recompute complexes
+    newValue = +newValue;
+
+    //determine if radius has increased or decreased
+    var radiusChange = 0;
+    if (newValue > complexRadius) {
+        radiusChange = 1;
+    } else if (newValue < complexRadius) {
+        radiusChange = -1;
+    }
+
+    //update slider value and/or text value
+    complexRadius = newValue;
     d3.select('#complexRadius').node().value =  complexRadius;
     d3.select('#complexInput').node().value = complexRadius;
 
-    //adjust inner and outer coverage disks
-    var innerRadius = xScale(complexRadius - dataRadius + xScale.domain()[0]);
-    var outerRadius = xScale(complexRadius + dataRadius + xScale.domain()[0]);
-    d3.select('#complexCircles').selectAll('circle').attr('r', innerRadius);
-    d3.select('#complexDataCircle').selectAll('circle').attr('r', outerRadius);
 
     //recompute complexes
     var t = Date.now();
-    constructRips();
+
+
+    if (radiusChange == 1) {
```

Comment:
Yeah, could be cleaned up a bit but not super important right now. I do ultimately plan on passing the radius change in as an argument so the algorithm knows which points to look at but this will be in a later version.